### PR TITLE
Convert Date format

### PR DESCRIPTION
**Pull request by kevinnorris**

### CODE REVIEW

**Comment by kevinnorris** [comment]
I used string concatenation instead of templating for deepEqual assert as I was getting a syntax error when using string template.

File: client/scripts/challenges/rosettacode/formatted/D/Date format.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const fmt1 = `${date.getFullYear()}-${(1 + date.getMonth())}-${date.getDate()}`;
+  const fmt2 = `${weekdays[date.getDay()]}, ${months[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
+  return [fmt1, fmt2];
+};
+
+/// tests:
+assert(typeof getDateFormats === 'function', 'message: <code>getDateFormats</code> is a function.');
+assert(typeof getDateFormats() === 'object', 'message: Should return an object.');
+assert(getDateFormats().length === 2, 'message: Should returned an array with 2 elements.');
+assert.deepEqual(getDateFormats(), getDateSolution(), 'message: <code>getDataFormats()</code> should return <code>' + getDateSolution().toString() + '</code>.');
```
Comment (tpoikela) [inline client/scripts/challenges/rosettacode/formatted/D/Date format.js]: In the error message in GUI it's not 100% clear that an array of 2 strings is required:
In the message:
`... should return 2017-7-13,Thursday, July 13, 2017`
To make it clear, change this to show 
`... should return ["2017-7-13", "Thursday, July 13, 2017"]`.

Comment (kevinnorris) [reply client/scripts/challenges/rosettacode/formatted/D/Date format.js]: Good catch

**Comment by tpoikela** [comment]
Regarding the templating, we have to investigate if we can make it work easily.

**Pull request merged**